output/statsd/output_test.go

```diff
@@ -0,0 +1,137 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2019 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package statsd
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/guregu/null.v3"
+
+	"github.com/loadimpact/k6/lib/testutils"
+	"github.com/loadimpact/k6/lib/types"
+	"github.com/loadimpact/k6/output"
+	"github.com/loadimpact/k6/stats"
+)
+
+func getOutput(
+	logger logrus.FieldLogger, addr, namespace null.String, bufferSize null.Int, pushInterval types.NullDuration,
+) (*Output, error) {
+	return newOutput(
+		output.Params{
+			Logger: logger,
+			JSONConfig: json.RawMessage(fmt.Sprintf(`{
+			"addr": "%s",
+			"namespace": "%s",
+			"bufferSize": %d,
+			"pushInterval": "%s"
+		}`, addr.String, namespace.String, bufferSize.Int64, pushInterval.Duration.String())),
+		})
+}
+
+func TestStatsdOutput(t *testing.T) {
+	t.Parallel()
+	baseTest(t, getOutput,
+		func(t *testing.T, _ []stats.SampleContainer, expectedOutput, output string) {
+			assert.Equal(t, expectedOutput, output)
+		})
+}
+
+func TestStatsdEnabledTags(t *testing.T) {
+	t.Parallel()
+	tagMap := stats.TagSet{"tag1": true, "tag2": true}
+
+	baseTest(t, func(
+		logger logrus.FieldLogger, addr, namespace null.String, bufferSize null.Int, pushInterval types.NullDuration,
+	) (*Output, error) {
+		return newOutput(
+			output.Params{
+				Logger: logger,
+				JSONConfig: json.RawMessage(fmt.Sprintf(`{
+			"addr": "%s",
+			"namespace": "%s",
+			"bufferSize": %d,
+			"pushInterval": "%s",
+			"tagBlocklist": ["tag1", "tag2"],
+			"enableTags": true
+		}`, addr.String, namespace.String, bufferSize.Int64, pushInterval.Duration.String())),
+			})
+	}, func(t *testing.T, containers []stats.SampleContainer, expectedOutput, output string) {
+		outputLines := strings.Split(output, "\n")
+		expectedOutputLines := strings.Split(expectedOutput, "\n")
+		for i, container := range containers {
+			for j, sample := range container.GetSamples() {
```

Comment:
Can you add a `require.Len()` assertion for both `containers` and `container.GetSamples()`? Otherwise if they're empty none of the checks below will run and the test will silently pass.